### DISCUSSION TITLE
Is it possible to integrate with a Framework DLL that has a hard dependency on System.Web.Hosting.HostingEnvironment?

### DISCUSSION THREAD

**Discussion by IanKemp** [dotnet/runtime#38859] [General] [answered]
I have to build an integration solution that talks to BIG_NAME_FINANCIAL_SYSTEM via their API, which takes the form of a .NET framework 4.5.2 DLL. The vendor is not interested in building or providing a web API, nor are they interested in updating their code to .NET Core.

My intention is to build a .NET Core background service that will pull data from/push data to this system, using their API. However I have essentially fallen at the first hurdle because `System.Web` in .NET Core is essentially a husk of what it was in Framework, and their API depends on Framework bits. In particular, it has a hard dependency on the `System.Web.Hosting.HostingEnvironment` class, which obviously doesn't exist in Core because the web pipeline was completely decoupled and rearchitected.

The end result is that when I run my Core app that references BIG_NAME_FINANCIAL_SYSTEM's API DLL, their DLL throws a `TypeLoadException` complaining it cannot find `System.Web.Hosting.HostingEnvironment` in `System.Web.dll`. Again, I understand why this happens, but is there any way to get around it?

Yes, I know I can build a Framework app that references their API DLL, then exposes its own API which my Core app can then talk to - but that's another layer of indirection, and latency, and project to maintain, and hence I'd prefer to avoid that if at all possible.

**Comment by davidfowl** [comment, marked as answer]
You can probably use this compat assembly that basically noops https://github.com/dotnet/corefxlab/tree/master/src/System.Web.Compatibility 

cc @ericstj

**Reply by ericstj** [reply]
Yep, try it out and let us know how it works for you.   See the instructions for connecting to the feed here:
https://dnceng.visualstudio.com/public/_packaging?_a=connect&feed=dotnet-experimental

Then reference the package
```xml
<PackageReference Include="System.Web.Compatibility" Version="0.0.1-alpha.20379.1" />
```
